src/Assert.php

```diff
@@ -467,8 +467,8 @@ public static function isAOf($value, $class, $message = '')
 
         if (!\is_a($value, $class, \is_string($value))) {
             static::reportInvalidArgument(sprintf(
-                $message ?: 'Expected an instance of this class or to this class among his parents %2$s. Got: %s',
-                static::typeToString($value),
+                $message ?: 'Expected an instance of this class or to this class among its parents "%2$s". Got: %s',
+                static::valueToString($value),
```

Comment:
I still feel that `typeToString()` is correct here. 

I want to see `string`, `AcmeService` or `boolean`. I don't want to see the string value. Do I?

Reply:
Well, I think it depends whether the value is passed as an object or string. WDYT about the following?
```php
$isValueAString = \is_string($value);

if (!\is_a($value, $class, $isValueAString)) {
    static::reportInvalidArgument(sprintf(
        $message ?: 'Expected an instance of this class or to this class among its parents "%2$s". Got: %s',
        $isValueAString ? static::valueToString($value) : static::typeToString($value),  
        $class
    ));
}
```
An assertion about checking the type of value whether it is an object or string might be also a good idea

Reply:
I updated the invalid test cases with passing an object and `static::valueToString($value)` is correct here.